osu.Game/Overlays/Chat/ChannelList/ChannelList.cs

```diff
@@ -0,0 +1,175 @@
+                    ScrollDistance = 35f,
+                    Child = new FillFlowContainer
+                    {
+                        Direction = FillDirection.Vertical,
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                        Children = new Drawable[]
+                        {
+                            publicChannelFlow = new ChannelListItemFlow("CHANNELS"),
+                            new ChannelListSelector
+                            {
+                                Margin = new MarginPadding { Bottom = 10 },
+                                SelectorActive = { BindTarget = SelectorActive },
+                            },
+                            privateChannelFlow = new ChannelListItemFlow("DIRECT MESSAGES"),
+                        },
+                    },
+                },
+            };
+        }
+
+        public void AddChannel(Channel channel)
+        {
+            if (channelMap.ContainsKey(channel))
+                return;
+
+            ChannelListItem item = new ChannelListItem(channel);
+            item.OnRequestSelect += chan => OnRequestSelect?.Invoke(chan);
+            item.OnRequestLeave += chan => OnRequestLeave?.Invoke(chan);
+
+            ChannelListItemFlow flow = getFlowForChannel(channel);
+            channelMap.Add(channel, item);
+            flow.Add(item);
+        }
+
+        public void RemoveChannel(Channel channel)
+        {
+            if (!channelMap.ContainsKey(channel))
+                return;
+
+            ChannelListItem item = channelMap[channel];
+            ChannelListItemFlow flow = getFlowForChannel(channel);
+
+            channelMap.Remove(channel);
+            flow.Remove(item);
+            item.Expire();
+        }
+
+        public ChannelListItem GetItem(Channel channel)
+        {
+            if (!channelMap.ContainsKey(channel))
+                throw new ArgumentOutOfRangeException();
+
+            return channelMap[channel];
+        }
+
+        private ChannelListItemFlow getFlowForChannel(Channel channel)
+        {
+            switch (channel.Type)
+            {
+                case ChannelType.Public:
+                    return publicChannelFlow;
+
+                case ChannelType.PM:
+                    return privateChannelFlow;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private class ChannelListItemFlow : FillFlowContainer
+        {
+            public ChannelListItemFlow(string label)
+            {
+                Direction = FillDirection.Vertical;
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+
+                Add(new OsuSpriteText
+                {
+                    Text = label,
+                    Margin = new MarginPadding { Left = 18, Bottom = 5 },
+                    Font = OsuFont.Torus.With(size: 12),
```

Comment:
Font sizes this small should come with a weight spec; this size is looking rather aliased for me right now. `FontWeight.SemiBold` should do.